core-plugins/docs/XMLReader-batchsource.md

```diff
@@ -0,0 +1,111 @@
+# XMLReader Batch Source
+
+
+Description
+-----------
+The XML Reader plugin is a source plugin that allows users to read XML files stored on HDFS.
+
+
+Use Case
+--------
+A user would like to read XML files that have been dropped into HDFS.
+These can be small to very large XML files. The XMLReader will read and parse the files,
+and when used in conjunction with the XMLParser plugin, fields can be extracted.
+This reader emits one XML event, specified by the node path property, for each file read.
+
+
+Properties
```

Comment:
There are no default values for nullable inputs, hence not mentioned in the description.
For optional fields, there  is no `@Nullable` in java class but how to mention this in document as I did not get any reference in existing documents for optional and mandatory fields.
Shall I put * after fields lable or put text as (Optional) afte the lable, please confirm?